src/main.ts

```diff
@@ -55,8 +54,8 @@ async function main() {
   // NB: Write empties the buffer must be after the comment.
   await summary.write();
   // Finally, set the status of the action.
-  for (const key of Object.keys(steps)) {
-    if (steps[key].status == Status.Failed) {
+  for (const [key, value] of Object.entries(steps)) {
```

Comment:
The `Object.entries` looses the type and goes to `string, any`.

Reply:
Yes, unfortunately typescript is representing javascript correctly here. Steps could contain other things. You could help it be explicit with something like:

```ts
 for (const [key, value] of Object.entries(steps) as [
    keyof Steps,
    Result | undefined
  ][]) {
```

Will suffice to make TS happy.

Reply:
Thanks! Added the type sig, but still don't get an error on `value.status` vs `value?.status`. Seems like the value doesn't respect the undefined or it makes no diff to the type check here?

Reply:
Hmm, that is weird, haven't seen it such that it coerced. Might be a TS option that I'm missing somewhere.